src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -239,33 +239,46 @@ public PatternMatches GetMatches(string candidate, string dottedContainer)
         /// so, unless you need to know the full set of matches, use this version.
         /// </remarks>
         /// <param name="candidate">The word being tested.</param>
-        /// <param name="inludeMatchSpans">Whether or not the matched spans should be included with results</param>
+        /// <param name="includeMatchSpans">Whether or not the matched spans should be included with results</param>
         /// <returns>If this was a match, the first element of the set of match types that occurred while matching the
         /// patterns. If it was not a match, it returns null.</returns>
-        public PatternMatch? GetFirstMatch(string candidate, bool inludeMatchSpans = false)
+        public PatternMatch? GetFirstMatch(
+            string candidate, bool includeMatchSpans)
         {
             if (SkipMatch(candidate))
             {
                 return null;
             }
 
-            return MatchPatternSegment(candidate, inludeMatchSpans, _fullPatternSegment, wantAllMatches: false, allMatches: out _, fuzzyMatch: false) ??
-                   MatchPatternSegment(candidate, inludeMatchSpans, _fullPatternSegment, wantAllMatches: false, allMatches: out _, fuzzyMatch: true);
+            return GetFirstMatchWorker(candidate, includeMatchSpans, fuzzyMatch: false) ??
+                   GetFirstMatchWorker(candidate, includeMatchSpans, fuzzyMatch: true);
         }
 
-        private StringBreaks GetWordSpans(string word)
+        public PatternMatch? GetFirstMatch(
+            string candidate, bool includeMatchSpans, bool fuzzyMatch)
         {
-            lock (_gate)
+            if (SkipMatch(candidate))
             {
-                return _stringToWordSpans.GetOrAdd(word, _breakIntoWordSpans);
+                return null;
             }
+
+            return GetFirstMatchWorker(candidate, includeMatchSpans, fuzzyMatch);
         }
 
-        internal PatternMatch? MatchSingleWordPattern_ForTestingOnly(string candidate)
+        private PatternMatch? GetFirstMatchWorker(
```

Comment:
:bulb: If the overloads of `GetFirstMatch` are unified, then this method can likely be inlined without hurting readability.